data/mods/gen3/scripts.ts

```diff
@@ -109,7 +190,9 @@ export const Scripts: ModdedBattleScriptsData = {
 				this.add('-notarget', pokemon);
 				return false;
 			}
-			if (targets.length > 1) move.spreadHit = true;
+			// In Generation 3, moves that hit both foes and the user's ally,
+			// like Earthquake and Explosion, don't get affected by spread modifiers
+			if (targets.length === 2) move.spreadHit = true;
```

Comment:
Doesn't this consider foe slots that are currently empty? I'd specifically check for unfainted slots and `move.target === 'allAdjacentFoes'` to be safe, then replace the animation check below just with `targets.length > 1` for good measure.